pyproject.toml

```diff
@@ -1,15 +1,65 @@
 [project]
 name = "faust-streaming"
 requires-python = ">=3.8"
-dynamic = ["version"]
+dynamic = [
+    "version",
+    "optional-dependencies",
+    "dependencies",
+
+]
+readme = "README.md"
+license = { file = "LICENSE" }
+keywords = [
+    "stream",
+    "processing",
+    "asyncio",
+    "distributed",
+    "queue",
+    "kafka",
+]
+classifiers = [
+    "Framework :: AsyncIO",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Natural Language :: English",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Operating System :: POSIX",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: POSIX :: BSD",
+    "Operating System :: Microsoft :: Windows",
+    "Topic :: System :: Networking",
+    "Topic :: System :: Distributed Computing",
+]
+
+authors = [
+#    "Ask Solem Hoel",
+]
+
+maintainers = [
+#    "William Barnhart <williambbarnhart@gmail.com>",
+#    "Vikram Patki",
+]
```

Comment:
Some reason these keep returning
```python
      OFFENDING RULE: 'type'
      
      DEFINITION:
          {
              "$id": "#/definitions/author",
              "title": "Author or Maintainer",
              "$comment": "https://peps.python.org/pep-0621/#authors-maintainers",
              "type": "object",
              "additionalProperties": false,
              "properties": {
                  "name": {
                      "type": "string",
                      "$$description": [
                          "MUST be a valid email name, i.e. whatever can be put as a name, before an",
                          "email, in :rfc:`822`."
                      ]
                  },
                  "email": {
                      "type": "string",
                      "format": "idn-email",
                      "description": "MUST be a valid email address"
                  }
              }
          }
      Traceback (most recent call last):
        File "/home/wbarnhart/PycharmProjects/faust/venv/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/home/wbarnhart/PycharmProjects/faust/venv/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/wbarnhart/PycharmProjects/faust/venv/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 132, in get_requires_for_build_editable
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 441, in get_requires_for_build_editable
          return self.get_requires_for_build_wheel(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 209, in <module>
        File "<string>", line 186, in do_setup
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 159, in setup
          dist.parse_config_files()
        File "/home/wbarnhart/PycharmProjects/faust/venv/lib/python3.11/site-packages/_virtualenv.py", line 21, in parse_config_files
          result = old_parse_config_files(self, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 627, in parse_config_files
          pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 66, in apply_configuration
          config = read_configuration(filepath, True, ignore_option_errors, dist)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 127, in read_configuration
          validate(subset, filepath)
        File "/tmp/pip-build-env-5bfe5sbh/overlay/lib/python3.11/site-packages/setuptools/config/pyprojecttoml.py", line 55, in validate
          raise ValueError(f"{error}\n{summary}") from None
      ValueError: invalid pyproject.toml config: `project.authors[0]`.
      configuration error: `project.authors[0]` must be object
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```
I'll need to look closer at this. I'm following an example on https://python-poetry.org/docs/pyproject/#maintainers but even the example no longer works. Did something change?

Reply:
> These keys accept an array of tables with 2 keys: name and email. Both values must be strings. The name value MUST be a valid email name (i.e. whatever can be put as a name, before an email, in [RFC 822](https://datatracker.ietf.org/doc/html/rfc822.html)) and not contain commas. The email value MUST be a valid email address. Both keys are optional, but at least one of the keys must be specified in the table.

I've tried all sorts of combinations and nothing seems to be compliant. Is there a bug that was introduced?

Reply:
Apparently it was updated in https://flit.pypa.io/en/stable/pyproject_toml.html.